x/delayedack/keeper/hooks.go

```diff
@@ -52,21 +52,18 @@ func (k Keeper) GetEpochHooks() epochstypes.EpochHooks {
 }
 
 // BeforeEpochStart is the epoch start hook.
-func (e epochHooks) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
-	return nil
-}
+func (e epochHooks) BeforeEpochStart(sdk.Context, string, int64) error { return nil }
 
 // AfterEpochEnd is the epoch end hook.
 // We want to clean up the demand orders that are with underlying packet status which are finalized.
-func (e epochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+func (e epochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64) error {
 	if epochIdentifier != e.GetParams(ctx).EpochIdentifier {
 		return nil
 	}
 	// Get all rollapp packets with status != PENDING and delete them
-	finalizedRollappPackets := e.ListRollappPacketsByStatus(ctx, commontypes.Status_FINALIZED, 0)
-	revertedRollappPackets := e.ListRollappPacketsByStatus(ctx, commontypes.Status_REVERTED, 0)
-	toDeletePackats := append(finalizedRollappPackets, revertedRollappPackets...)
-	for _, packet := range toDeletePackats {
+	toDeletePackets := e.ListRollappPackets(ctx, ByNotStatus(commontypes.Status_PENDING))
```

Comment:
I’m afraid it will be very inefficient as we need to iterate over all packets every interval where the majority will be pending packets assuming epoch > grace period. 
Original epoch planned was 1hr.

Reply:
We have one packet key which starts with the rollappID now. My idea of narrowing down the number of packets for most use cases is if we first filter by rollappID prefix and then by status. If we get a subset of packets by RollApp, I wasn't sure, but I thought it might be smaller than a subset by status - FINALIZED + REVERTED.
As it was, we fetch all FINALIZED and REVERTED rollapps and then loop over them. 

I'll think of a better way